unfoldingWord/translationCore
# Improves builds

#### This pull request addresses:

This changes the build process to use electron-packager directly instead of via electron-forge.
Doing so allows us to have more control over compiling installers.

In addition to changes in the build process:

* the windows installer now uses a native looking installation dialog with prompts.
* the macOS application is now distributed in a .dmg archive. However, the dmg is currently uncompressed. I hope to figure out a way to compress it later.
* the macOS app closes down completely when the window is closed. Previously it would remain running with no way to re-display the window, requiring the user to force shut down the app.

This also includes some fixes to `myProjectsHelpers` which was breaking the CSVExportActions tests.

#### How to test this pull request:

The generated installers for Windows and macOS should run properly.

The easiest way to generate *all* installers is on linux. On windows you can generate the windows installers by first installing [Inno Setup](http://www.jrsoftware.org/isdl.php). You need this software on linux as well, however there's a script to install it automatically in `scripts/innosetup/setup.sh`

Generating the installers is a two step process: build (packages code), then release (prepares installer/distributable). See the travis config file for examples.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/unfoldingword-dev/translationcore/3146)
